templates/CRM/PCP/Form/Search.tpl

```diff
@@ -0,0 +1,223 @@
+                       "sProcessing":    {/literal}"{ts escape='js'}Processing...{/ts}"{literal},
+                       "sLengthMenu":    {/literal}"{ts escape='js'}Show _MENU_ entries{/ts}"{literal},
+                       "sInfo":          {/literal}"{ts escape='js'}Showing _START_ to _END_ of _TOTAL_ entries{/ts}"{literal},
+                       "sInfoEmpty":     {/literal}"{ts escape='js'}Showing 0 to 0 of 0 entries{/ts}"{literal},
+                       "sInfoFiltered":  {/literal}"{ts escape='js'}(filtered from _MAX_ total entries){/ts}"{literal},
+                       "sSearch":        {/literal}"{ts escape='js'}Search:{/ts}"{literal},
+                       "oPaginate": {
+                            "sFirst":    {/literal}"{ts escape='js'}First{/ts}"{literal},
+                            "sPrevious": {/literal}"{ts escape='js'}Previous{/ts}"{literal},
+                            "sNext":     {/literal}"{ts escape='js'}Next{/ts}"{literal},
+                            "sLast":     {/literal}"{ts escape='js'}Last{/ts}"{literal}
+                        }
+                    },
+        "fnRowCallback": function(nRow, aData, iDisplayIndex, iDisplayIndexFull) {
+	  var status = $('.crm-pcp-status', nRow).text();
+	  var action = '';
+	  if(status=='Approved'){
+	    action = 'revert';
+	  } else {
+	    action = 'renew';
+	  }
+	  var id = $('td:last', nRow).text().split(',')[0];
+          var cl = $('td:last', nRow).text().split(',')[1];
+          $(nRow).addClass(cl).attr({id: 'row_' + id, 'data-id': id, 'data-entity': 'PCP', 'action': action});
+          return nRow;
+        },
+        "fnDrawCallback": function() {
+          // FIXME: trigger crmLoad and crmEditable would happen automatically
+          $('.crm-editable').crmEditable();
+        },
+        "fnServerData": function ( sSource, aoData, fnCallback ) {
+            if ( filterSearch ) {
+                aoData.push(
+                    {name:'title', value: $('.crm-pcp-search-form-block #title').val()},
+                    {name:'page_type', value: $('.crm-pcp-search-form-block #page_type').val()},
+                    {name:'page_id', value: $('.crm-pcp-search-form-block #page_id').val()},
+                    {name:'event_id', value: $('.crm-pcp-search-form-block #event_id').val()},
+                    {name:'status_id', value: $('.crm-pcp-search-form-block #status_id').val()},
+                    {name:'supporter', value: $('.crm-pcp-search-form-block #supporter').val()},
+                    {name:'start_date', value: $('.crm-pcp-search-form-block #start_date').val()},
+                    {name:'end_date', value: $('.crm-pcp-search-form-block #end_date').val()}
+                );
+            }
+            $.ajax( {
+                "dataType": 'json',
+                "type": "POST",
+                "url": sSource,
+                "data": aoData,
+                "success": fnCallback
+            } );
+        }
+    });
```

Comment:
All of the above seems to be using the legacy 1.0 datatables api code.
Can you use the modern approach to datatables we have adopted for 4.7?
See https://issues.civicrm.org/jira/browse/CRM-16353